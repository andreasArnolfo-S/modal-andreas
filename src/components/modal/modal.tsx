import React, { FC } from 'react';
import styles from './modal.module.css';

interface ModalProps {}

const Modal: FC<ModalProps> = () => (
  <div className={styles.Modal}>
    Modal Component
  </div>
);

export default Modal;
