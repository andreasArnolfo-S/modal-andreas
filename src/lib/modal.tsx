import * as React from "react";

interface ModalProps {
     isOpen: boolean;
     contentTitle: string;
     contentBody: string;
     success?: boolean;
     modalSize: string;
     styleGlass?: boolean;
     onClose: () => void;
     btnLabel?: string;
     btnColor?: string;
     children?: React.ReactNode;
     isClosable?: boolean;
     type?: string;
}


export const Modal: React.FC<ModalProps> = ({ isOpen, isClosable, contentBody, type, contentTitle, success, onClose, modalSize, styleGlass, btnColor, btnLabel, children }) => {
     React.useEffect(() => {
          const handleKeyDown = (event: KeyboardEvent) => {
               if (event.key === 'Escape' && isClosable) {
                    onClose();
               }
          };
          window.addEventListener('keydown', handleKeyDown);
          return () => {
               window.removeEventListener('keydown', handleKeyDown);
          };
     }, [isClosable, onClose]);


     return (
          <>

               {isOpen ?
                    <div data-test-id='modal' style={{ position: 'fixed', top: 0, left: 0, width: '100%', height: '100%', backgroundColor: 'rgba(0,0,0,0.5)', display: 'flex', justifyContent: 'center', alignItems: 'center', zIndex: 10000}}>
                         <div style={{ width: `${modalSize}`, backgroundColor: `${success ? '#fff' : '#deaeab'}`, borderRadius: '0.5rem', boxShadow: '0 0 10px 0 rgba(0,0,0,0.2)', padding: '1rem', zIndex: 10, position: 'relative', }}>
                              <h3 style={{ fontWeight: 700, fontSize: '1.5rem', lineHeight: '2rem', textAlign: 'center', }}>
                                   {contentTitle}
                              </h3>
                              <p style={{ paddingTop: '1rem', paddingBottom: '1rem', textAlign: 'center', }}>
                                   {contentBody}
                              </p>
                              {type === 'componentInside' ? <div style={{ paddingTop: '1rem', paddingBottom: '1rem' }}>
                                   {children}
                              </div> : null}
                              {isClosable ?
                                   <div className="closeTopRight" style={{ position: 'absolute', top: '-0.5rem', right: '-0.6rem', zIndex: 10, display: 'flex', justifyContent: 'center', alignItems: 'center', }}>
                                        <button data-testid="closeButton" onClick={onClose} style={{backgroundColor: `${btnColor}`,border: 'none',cursor: 'pointer',fontSize: '1rem',width: 30,height: 30,textAlign: 'center',color: '#fff',borderRadius: '50%',}}>
                                             x
                                        </button>
                                   </div> : null}
                              {type === 'confirmation' ? <div style={{display: 'flex', justifyContent: 'center', gap: '2%'}}>
                                   <button data-testid='okcButton' style={{backgroundColor: 'green', borderRadius: '5px', padding: '10px 20px', fontSize: '16px', cursor: 'pointer', border: 'none'}}>Ok</button>
                                   <button data-testid='cancelButton' style={{backgroundColor: 'gray', borderRadius: '5px', padding: '10px 20px', fontSize: '16px', cursor: 'pointer', border: 'none'}}>Cancel</button>
                              </div> : null}

                              {type === 'formulaire' ? 
                              //form
                              <form style={{display: 'flex', flexDirection: 'column', gap: '2%', marginBottom: '15px'}}>
                                   <label htmlFor="Nom">Nom</label>
                                   <input type="text" placeholder="Nom" style={{padding: '10px 20px', borderRadius: '5px', border: 'none', fontSize: '16px'}}/>
                                   <label htmlFor="Prénom">Prénom</label>
                                   <input type="text" placeholder="Prénom" style={{padding: '10px 20px', borderRadius: '5px', border: 'none', fontSize: '16px'}}/>
                                   <label htmlFor="Email">Email</label>
                                   <input type="text" placeholder="Email" style={{padding: '10px 20px', borderRadius: '5px', border: 'none', fontSize: '16px'}}/>
                                   <label htmlFor="Téléphone">Téléphone</label>
                                   <input type="text" placeholder="Téléphone" style={{padding: '10px 20px', borderRadius: '5px', border: 'none', fontSize: '16px'}}/>
                                   <button style={{backgroundColor: 'green', borderRadius: '5px', padding: '10px 20px', fontSize: '16px', cursor: 'pointer', border: 'none', marginTop: '15px'}}>Ok</button>
                              </form> : null}
                         </div>
                    </div> : null
               }
          </>
     );
}
