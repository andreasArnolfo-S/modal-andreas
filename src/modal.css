.fixed {
     position: fixed;
     top: 0;
     left: 0;
     width: 100%;
     height: 100%;
     z-index: 1000;
     background-color: rgba(161, 161, 161, 0.5);
}

.modal-box {
     position: fixed;
     top: 50%;
     left: 50%;
     transform: translate(-50%, -50%);
     width: 400px;
     height: 300px;
     background-color: #fff;
     z-index: 1001;
     border-radius: 15px;
     padding: 20px;
     box-shadow: 0 25px 50px -12px rgba(0, 0, 0, 0.25);
}
.text-2xl {
     font-size: 1.5rem/* 24px */;
     line-height: 2rem/* 32px */;
}
.font-bold {
     font-weight: 700;
}

.btn {
     background-color: #fff;
     border: 1px solid #000;
     border-radius: 5px;
     padding: 10px 20px;
     font-size: 16px;
     cursor: pointer;
     margin-bottom: 15px;
}
.btn:hover {
     background-color: rgb(49, 49, 49);
     color: #fff;
}

.btn-primary {
     background-color: #000;
     color: #fff;
}
.btn-secondary {
     background-color: #fff;
     color: #000;
}
.btn-danger {
     background-color: #f00;
     color: #fff;
}
.btn-accent {
     background-color: #f0f;
     color: #000;
}
.btn-success {
     background-color: #0f0;
     color: #000;
}

.py-4 {
     padding-top: 1.5rem;
     padding-bottom: 1.5rem;
}

.flex {
     display: flex;
}

.flex-col {
     flex-direction: column;
}
.modal-action {
     display: flex;
     margin-top: 1.5rem/* 24px */;
     justify-content: flex-end;
}

.text-center {
     text-align: center;
}

.justify-center {
     justify-content: center;
}