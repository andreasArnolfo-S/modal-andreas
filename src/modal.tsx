import * as React from "react";

interface ModalProps {
     isOpen: boolean;
     contentTitle: string;
     contentBody: string;
     success: boolean;
     user?: {
          firstName: string,
          lastName: string,
          startDate: string,
          city: string,
          department: string,
     },
     onClose: () => void;
}

export const Modal: React.FC<ModalProps> = ({ isOpen, contentBody, contentTitle, success, onClose, user }) => {
     return (
          <>
               <link href="https://cdn.jsdelivr.net/npm/daisyui@2.46.1/dist/full.css" rel="stylesheet" type="text/css" />
               <link href="https://cdn.jsdelivr.net/npm/tailwindcss@2.2/dist/tailwind.min.css" rel="stylesheet" type="text/css" />

               {isOpen ?
                    <div className="fixed top-0 right-0 left-0 bottom-0 flex justify-center duration-200 ease-in-out bg-base-300">
                         <div className="modal-box w-11/12 max-w-6xl h-2/3 flex justify-center flex-col">
                              <h3 className="font-bold text-2xl text-center">{contentTitle}</h3>
                              <p className="py-4 text-center">{contentBody}</p>

                              <div className="avatar placeholder justify-center mb-15">
                                   <div className="bg-neutral-focus text-neutral-content rounded-full w-16">
                                        <span className="text-xl">JL</span>
                                   </div>
                              </div>

                              <div className="stats shadow">

                                   <div className="stat">
                                        <div className="stat-title">First name</div>
                                        <div className="stat-value text-primary">{user?.firstName}</div>
                                   </div>

                                   <div className="stat">
                                        <div className="stat-title">Last name</div>
                                        <div className="stat-value text-secondary">{user?.lastName}</div>
                                   </div>

                                   <div className="stat">
                                        <div className="stat-title">Start date</div>
                                        <div className="stat-value text-accent">{user?.startDate}</div>
                                   </div>

                                   <div className="stat">
                                        <div className="stat-title">Department</div>
                                        <div className="stat-value text-warning">{user?.department}</div>
                                   </div>

                                   <div className="stat">
                                        <div className="stat-title">City</div>
                                        <div className="stat-value text-base-300">{user?.city}</div>
                                   </div>

                                   

                              </div>

                              <div className="modal-action justify-center">
                                   <button onClick={onClose} className="btn">Close</button>
                              </div>
                         </div>
                    </div> : null
               }
          </>
     );
}
