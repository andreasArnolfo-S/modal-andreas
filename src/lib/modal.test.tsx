import * as React from "react";
import { Modal } from "./modal";
import userEvent from '@testing-library/user-event'
import { render, screen, getAllByTestId, getByTestId, fireEvent } from '@testing-library/react';


describe('modal', () => {

  it('should render modal with default type', () => {
    const { container } = render(<Modal isOpen={true} contentTitle="Modal Title" success={true} contentBody="Modal Body" modalSize='600px' onClose={jest.fn()} />);
    expect(container).toMatchSnapshot();
  });

  it('click on close button should close the modal', () => {
    const { container } = render(<Modal isOpen={true} contentTitle="Modal Title" success={true} contentBody="Modal Body" modalSize='600px' onClose={jest.fn()} isClosable={true} />);
    const closeButton = screen.getByTestId('closeButton');
    userEvent.click(closeButton);
    expect(container).toMatchSnapshot();
  });

  it('press ESC key should close the modal', () => {
    const { container } = render(<Modal isOpen={true} contentTitle="Modal Title" success={true} contentBody="Modal Body" modalSize='600px' onClose={jest.fn()} isClosable={true} />);
    fireEvent.keyDown(document, { key: 'Escape', code: 'Escape' });
    expect(container).toMatchSnapshot();
  });

  it('should render modal with type success', () => {
    const { container } = render(<Modal isOpen={true} contentTitle="Modal Title" success={true} contentBody="Modal Body" modalSize='600px' onClose={jest.fn()} type='success' />);
    expect(container).toMatchSnapshot();
  });

  it('should render modal with type comfirmation', () => {
    const { container } = render(<Modal isOpen={true} contentTitle="Modal Title" success={true} contentBody="Modal Body" modalSize='600px' onClose={jest.fn()} type='comfirmation' />);
    expect(container).toMatchSnapshot();
  });

  it('should render modal with type formulaire', () => {
    const { container } = render(<Modal isOpen={true} contentTitle="Modal Title" success={true} contentBody="Modal Body" modalSize='600px' onClose={jest.fn()} type='formulaire' />);
    expect(container).toMatchSnapshot();
  });

  it('should render modal with type componentInside', () => {
    const { container } = render(<Modal isOpen={true} contentTitle="Modal Title" success={true} contentBody="Modal Body" modalSize='600px' onClose={jest.fn()} type='componentInside'><div>test</div></Modal>);
    expect(container).toMatchSnapshot();
  });


});
