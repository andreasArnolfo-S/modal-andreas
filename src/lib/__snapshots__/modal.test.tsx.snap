// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal click on close button should close the modal 1`] = `
<div>
  <div
    data-test-id="modal"
    style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; background-color: rgba(0, 0, 0, 0.5); display: flex; justify-content: center; align-items: center; z-index: 10000;"
  >
    <div
      style="width: 600px; background-color: rgb(255, 255, 255); border-radius: 0.5rem; box-shadow: 0 0 10px 0 rgba(0,0,0,0.2); padding: 1rem; z-index: 10; position: relative;"
    >
      <h3
        style="font-weight: 700; font-size: 1.5rem; line-height: 2rem; text-align: center;"
      >
        Modal Title
      </h3>
      <p
        style="padding-top: 1rem; padding-bottom: 1rem; text-align: center;"
      >
        Modal Body
      </p>
      <div
        class="closeTopRight"
        style="position: absolute; top: -0.5rem; right: -0.6rem; z-index: 10; display: flex; justify-content: center; align-items: center;"
      >
        <button
          data-testid="closeButton"
          style="cursor: pointer; font-size: 1rem; width: 30px; height: 30px; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
        >
          x
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`modal press ESC key should close the modal 1`] = `
<div>
  <div
    data-test-id="modal"
    style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; background-color: rgba(0, 0, 0, 0.5); display: flex; justify-content: center; align-items: center; z-index: 10000;"
  >
    <div
      style="width: 600px; background-color: rgb(255, 255, 255); border-radius: 0.5rem; box-shadow: 0 0 10px 0 rgba(0,0,0,0.2); padding: 1rem; z-index: 10; position: relative;"
    >
      <h3
        style="font-weight: 700; font-size: 1.5rem; line-height: 2rem; text-align: center;"
      >
        Modal Title
      </h3>
      <p
        style="padding-top: 1rem; padding-bottom: 1rem; text-align: center;"
      >
        Modal Body
      </p>
      <div
        class="closeTopRight"
        style="position: absolute; top: -0.5rem; right: -0.6rem; z-index: 10; display: flex; justify-content: center; align-items: center;"
      >
        <button
          data-testid="closeButton"
          style="cursor: pointer; font-size: 1rem; width: 30px; height: 30px; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
        >
          x
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`modal should render modal with default type 1`] = `
<div>
  <div
    data-test-id="modal"
    style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; background-color: rgba(0, 0, 0, 0.5); display: flex; justify-content: center; align-items: center; z-index: 10000;"
  >
    <div
      style="width: 600px; background-color: rgb(255, 255, 255); border-radius: 0.5rem; box-shadow: 0 0 10px 0 rgba(0,0,0,0.2); padding: 1rem; z-index: 10; position: relative;"
    >
      <h3
        style="font-weight: 700; font-size: 1.5rem; line-height: 2rem; text-align: center;"
      >
        Modal Title
      </h3>
      <p
        style="padding-top: 1rem; padding-bottom: 1rem; text-align: center;"
      >
        Modal Body
      </p>
    </div>
  </div>
</div>
`;

exports[`modal should render modal with type comfirmation 1`] = `
<div>
  <div
    data-test-id="modal"
    style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; background-color: rgba(0, 0, 0, 0.5); display: flex; justify-content: center; align-items: center; z-index: 10000;"
  >
    <div
      style="width: 600px; background-color: rgb(255, 255, 255); border-radius: 0.5rem; box-shadow: 0 0 10px 0 rgba(0,0,0,0.2); padding: 1rem; z-index: 10; position: relative;"
    >
      <h3
        style="font-weight: 700; font-size: 1.5rem; line-height: 2rem; text-align: center;"
      >
        Modal Title
      </h3>
      <p
        style="padding-top: 1rem; padding-bottom: 1rem; text-align: center;"
      >
        Modal Body
      </p>
    </div>
  </div>
</div>
`;

exports[`modal should render modal with type componentInside 1`] = `
<div>
  <div
    data-test-id="modal"
    style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; background-color: rgba(0, 0, 0, 0.5); display: flex; justify-content: center; align-items: center; z-index: 10000;"
  >
    <div
      style="width: 600px; background-color: rgb(255, 255, 255); border-radius: 0.5rem; box-shadow: 0 0 10px 0 rgba(0,0,0,0.2); padding: 1rem; z-index: 10; position: relative;"
    >
      <h3
        style="font-weight: 700; font-size: 1.5rem; line-height: 2rem; text-align: center;"
      >
        Modal Title
      </h3>
      <p
        style="padding-top: 1rem; padding-bottom: 1rem; text-align: center;"
      >
        Modal Body
      </p>
      <div
        style="padding-top: 1rem; padding-bottom: 1rem;"
      >
        <div>
          test
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`modal should render modal with type formulaire 1`] = `
<div>
  <div
    data-test-id="modal"
    style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; background-color: rgba(0, 0, 0, 0.5); display: flex; justify-content: center; align-items: center; z-index: 10000;"
  >
    <div
      style="width: 600px; background-color: rgb(255, 255, 255); border-radius: 0.5rem; box-shadow: 0 0 10px 0 rgba(0,0,0,0.2); padding: 1rem; z-index: 10; position: relative;"
    >
      <h3
        style="font-weight: 700; font-size: 1.5rem; line-height: 2rem; text-align: center;"
      >
        Modal Title
      </h3>
      <p
        style="padding-top: 1rem; padding-bottom: 1rem; text-align: center;"
      >
        Modal Body
      </p>
      <form
        style="display: flex; flex-direction: column; gap: 2%; margin-bottom: 15px;"
      >
        <label
          for="Nom"
        >
          Nom
        </label>
        <input
          placeholder="Nom"
          style="padding: 10px 20px; border-radius: 5px; font-size: 16px;"
          type="text"
        />
        <label
          for="Prénom"
        >
          Prénom
        </label>
        <input
          placeholder="Prénom"
          style="padding: 10px 20px; border-radius: 5px; font-size: 16px;"
          type="text"
        />
        <label
          for="Email"
        >
          Email
        </label>
        <input
          placeholder="Email"
          style="padding: 10px 20px; border-radius: 5px; font-size: 16px;"
          type="text"
        />
        <label
          for="Téléphone"
        >
          Téléphone
        </label>
        <input
          placeholder="Téléphone"
          style="padding: 10px 20px; border-radius: 5px; font-size: 16px;"
          type="text"
        />
        <button
          style="background-color: green; border-radius: 5px; padding: 10px 20px; font-size: 16px; cursor: pointer; margin-top: 15px;"
        >
          Ok
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`modal should render modal with type success 1`] = `
<div>
  <div
    data-test-id="modal"
    style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; background-color: rgba(0, 0, 0, 0.5); display: flex; justify-content: center; align-items: center; z-index: 10000;"
  >
    <div
      style="width: 600px; background-color: rgb(255, 255, 255); border-radius: 0.5rem; box-shadow: 0 0 10px 0 rgba(0,0,0,0.2); padding: 1rem; z-index: 10; position: relative;"
    >
      <h3
        style="font-weight: 700; font-size: 1.5rem; line-height: 2rem; text-align: center;"
      >
        Modal Title
      </h3>
      <p
        style="padding-top: 1rem; padding-bottom: 1rem; text-align: center;"
      >
        Modal Body
      </p>
    </div>
  </div>
</div>
`;
